### PR TITLE
ST: Add test which do downgrade to older log message format

### PR DESCRIPTION
**Pull request by Frawless**
Signed-off-by: Jakub Stejskal <xstejs24@gmail.com>

### Type of change

- Enhancement / new feature

### Description
This PR adds ST coverage for https://github.com/strimzi/strimzi-kafka-operator/pull/3639

### Checklist

- [x] Write tests
- [x] Make sure all tests pass

### CODE REVIEW

**Comment by Frawless** [comment]
@strimzi-ci  run tests profile=upgrade testcase=ZookeeperUpgradeST test_only

File: systemtest/src/test/java/io/strimzi/systemtest/upgrade/ZookeeperUpgradeST.java [hunk 1]
```diff
@@ -52,7 +52,16 @@ void testKafkaClusterUpgrade(ExtensionContext testContext) {
...
             TestKafkaVersion newVersion = sortedVersions.get(x + 1);
 
-            runVersionChange(initialVersion, newVersion, 3, 3, testContext);
+            String logMsgFormat;
+            if (initialVersion.compareTo(newVersion) < 0) {
+                // If it is an upgrade test we keep the message format as the lower version number
+                logMsgFormat = initialVersion.messageVersion();
+            } else {
+                // If it is a downgrade then we make sure to use the lower version number for the message format
+                logMsgFormat = newVersion.messageVersion();
```
Comment (see-quick) [inline systemtest/src/test/java/io/strimzi/systemtest/upgrade/ZookeeperUpgradeST.java:61]:
```suggestion
           String logMsgFormat = initialVersion.compareTo(newVersion) < 0 ? 
           // If it is an upgrade test we keep the message format as the lower version number
           initialVersion.messageVersion() : 
           // If it is a downgrade then we make sure to use the lower version number for the message format
           newVersion.messageVersion()
```

File: systemtest/src/test/java/io/strimzi/systemtest/upgrade/ZookeeperUpgradeST.java [hunk 2]
```diff
@@ -70,7 +79,16 @@ void testKafkaClusterDowngrade(ExtensionContext testContext) {
...
+            }
```
Comment (see-quick) [inline systemtest/src/test/java/io/strimzi/systemtest/upgrade/ZookeeperUpgradeST.java:89]: same with that

**Review by see-quick — COMMENTED** [review]
Looks good, just one suggestion.

**Comment by Frawless** [comment]
@strimzi-ci run tests profile=upgrade testcase=ZookeeperUpgradeST test_only

**Review by scholzj — APPROVED** [review]

**Review by see-quick — APPROVED** [review]

**Pull request merged**